tests/components/weheat/snapshots/test_binary_sensor.ambr

```diff
@@ -0,0 +1,235 @@
+  EntityRegistryEntrySnapshot({
+    'aliases': set({
+    }),
+    'area_id': None,
+    'capabilities': None,
+    'config_entry_id': <ANY>,
+    'device_class': None,
+    'device_id': <ANY>,
+    'disabled_by': None,
+    'domain': 'binary_sensor',
+    'entity_category': None,
+    'entity_id': 'binary_sensor.test_model_indoor_unit_electric_heater',
+    'has_entity_name': True,
+    'hidden_by': None,
+    'icon': None,
+    'id': <ANY>,
+    'labels': set({
+    }),
+    'name': None,
+    'options': dict({
+    }),
+    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
+    'original_icon': None,
+    'original_name': 'Indoor unit electric heater',
+    'platform': 'weheat',
+    'previous_unique_id': None,
+    'supported_features': 0,
+    'translation_key': 'indoor_unit_electric_heater_state',
+    'unique_id': '0000-1111-2222-3333_indoor_unit_electric_heater_state',
+    'unit_of_measurement': None,
+  })
+# ---
+# name: test_binary_entities[binary_sensor.test_model_indoor_unit_electric_heater-state]
+  StateSnapshot({
+    'attributes': ReadOnlyDict({
+      'device_class': 'running',
+      'friendly_name': 'Test Model Indoor unit electric heater',
+    }),
+    'context': <ANY>,
+    'entity_id': 'binary_sensor.test_model_indoor_unit_electric_heater',
+    'last_changed': <ANY>,
+    'last_reported': <ANY>,
+    'last_updated': <ANY>,
+    'state': 'on',
+  })
+# ---
+# name: test_binary_entities[binary_sensor.test_model_indoor_unit_gas_boiler_heating_allowed-entry]
+  EntityRegistryEntrySnapshot({
+    'aliases': set({
+    }),
+    'area_id': None,
+    'capabilities': None,
+    'config_entry_id': <ANY>,
+    'device_class': None,
+    'device_id': <ANY>,
+    'disabled_by': None,
+    'domain': 'binary_sensor',
+    'entity_category': None,
+    'entity_id': 'binary_sensor.test_model_indoor_unit_gas_boiler_heating_allowed',
+    'has_entity_name': True,
+    'hidden_by': None,
+    'icon': None,
+    'id': <ANY>,
+    'labels': set({
+    }),
+    'name': None,
+    'options': dict({
+    }),
+    'original_device_class': None,
+    'original_icon': None,
+    'original_name': 'Indoor unit gas boiler heating allowed',
+    'platform': 'weheat',
+    'previous_unique_id': None,
+    'supported_features': 0,
+    'translation_key': 'indoor_unit_gas_boiler_state',
+    'unique_id': '0000-1111-2222-3333_indoor_unit_gas_boiler_state',
+    'unit_of_measurement': None,
+  })
+# ---
+# name: test_binary_entities[binary_sensor.test_model_indoor_unit_gas_boiler_heating_allowed-state]
+  StateSnapshot({
+    'attributes': ReadOnlyDict({
+      'friendly_name': 'Test Model Indoor unit gas boiler heating allowed',
+    }),
+    'context': <ANY>,
+    'entity_id': 'binary_sensor.test_model_indoor_unit_gas_boiler_heating_allowed',
+    'last_changed': <ANY>,
+    'last_reported': <ANY>,
+    'last_updated': <ANY>,
+    'state': 'unknown',
```

Comment:
To double check, when would this be unknown?

Reply:
I think the state can be unknown when no indoor unit is available, or depending on the type of indoor unit (with one or more heatpumps available). But @jesperraemaekers probably knows that best.

Reply:
Ideally we only create those entities when it is the case though, otherwise people are left with entities that are always unknown

Reply:
Any tips how to do that?

Reply:
Okay I guess only add them when the state is not None in the setup. @jesperraemaekers is the state only null/unknown when they're not available? Or can it change at runtime?

Reply:
How can we detect at the device if a field is present? If the case is that if the device supports it, it will either return True or False, we can just filter out the values where it returns `None`. But if the other fields can also return `None` while present, we can't rely on that

Reply:
Okay I filtered out the unknown values.